MUXSDKStats/MUXSDKStats/MUXSDKPlayerBinding.m

```diff
@@ -82,12 +82,97 @@ - (void)attachAVPlayer:(AVPlayer *)player {
               forKeyPath:@"currentItem"
                  options:NSKeyValueObservingOptionInitial | NSKeyValueObservingOptionNew
                  context:MUXSDKAVPlayerCurrentItemObservationContext];
+    _lastMediaRequest = 0;
+    _lastMediaRequestBytes = 0;
+    _lastErrorLogEventCount = 0;
+    _lastTransferEventCount = 0;
+    _lastTransferDuration= 0;
+    _lastTransferredBytes = 0;
+    //_minBitrate = 0;
+}
+
+-(NSString *)getHostName:(NSString *)urlString {
+    NSURL* url = [NSURL URLWithString:urlString];
+    return [url host];
 }
 
 - (void)timeUpdateTimer:(NSTimer *)timer {
     if (![self isTryingToPlay] && ![self isBuffering]) {
         [self dispatchTimeUpdateFromTimer];
     }
+
+    if (_player.currentItem != nil) {
+        AVPlayerItemAccessLog *log = _player.currentItem.accessLog;
+        if (log != nil && log.events.count > 0) {
+            // https://developer.apple.com/documentation/avfoundation/avplayeritemaccesslogevent?language=objc
+            AVPlayerItemAccessLogEvent *event = log.events[log.events.count - 1];
+
+            if (event.numberOfMediaRequests > 0 && event.numberOfMediaRequests != _lastMediaRequest) {
+                if (_lastTransferEventCount != log.events.count) {
+                    _lastTransferDuration= 0;
+                    _lastTransferredBytes = 0;
+                    _lastTransferEventCount = log.events.count;
+                }
+                //double transferDuration = event.transferDuration - _lastTransferDuration;
+                //long long transferredBytes = event.numberOfBytesTransferred - _lastTransferredBytes;
+                // !!! event.observedMinBitrate, event.observedMaxBitrate, event.observedBitrate don't seem to be accurate
+                // we did a chalres proxy dump try to calculate the bitrate, and compared with above values. It doesn't match
+                // but it matches with the calculation between line 141 to line 149
+
+                // Keep the next line of print for future reseaerch.
+                //NSLog(@"Bwm transferDuration %f, numOfBytesXferred %llu, min %fbps, max %fbps, observered %fbps", event.transferDuration, event.numberOfBytesTransferred, event.observedMinBitrate, event.observedMaxBitrate, event.observedBitrate);
+
+                NSString *domain = [self getHostName:event.URI];
+                MUXSDKBandwidthMetricData *loadData = [[MUXSDKBandwidthMetricData alloc] init];
+                loadData.requestType = @"ManifestOrMedia";
+                loadData.requestStart = [NSNumber numberWithLong: _lastTransferDuration * 1000];
```

Comment:
This isn't really going to be accurate, is it? I wonder... is there a way that we can get within 50 ms of accuracy, given that our timer is run every 50 ms? I'm thinking... if we use the wallclock time within the timer callback, and check when the number of requests has changed, we might have an _idea_ of when the request started, and then an idea of when the request ended.

These values work, since you're essentially patching the transferDuration into the time, but it's not correct in terms of wall-clock time, right?